File: GeminiAPI.cs
using System.Collections;
using UnityEngine;
using UnityEngine.Networking;

public class GeminiAPI : MonoBehaviour
{
    private string apiKey = "AIzaSyAkAQSpayk-vkq7RJ6RlJvUhcFh4DP6KbY"; // Your API key here
    private string model = "models/gemini-pro"; // Or your specific model name

    void Start()
    {
        StartCoroutine(GenerateText());
    }

    IEnumerator GenerateText()
    {
        string url = $"https://generativelanguage.googleapis.com/v1beta/models/{model}:generateText?key={apiKey}";
        string jsonData = "{\"prompt\":\"Hello, Gemini!\"}";

        UnityWebRequest request = new UnityWebRequest(url, "POST");
        byte[] bodyRaw = System.Text.Encoding.UTF8.GetBytes(jsonData);
        request.uploadHandler = new UploadHandlerRaw(bodyRaw);
        request.downloadHandler = new DownloadHandlerBuffer();
        request.SetRequestHeader("Content-Type", "application/json");

        yield return request.SendWebRequest();

        if (request.result == UnityWebRequest.Result.Success)
        {
            Debug.Log("Response: " + request.downloadHandler.text);
        }
        else
        {
            Debug.Log("Request failed: " + request.error);
        }
    }
}

File: gemini_api.py
from flask import Flask, request, jsonify
import google.generativeai as genai
import os

app = Flask(__name__)

# Set up Gemini API
genai.configure(api_key="AIzaSyAkAQSpayk-vkq7RJ6RlJvUhcFh4DP6KbY")  # Replace with your actual API key
model = genai.GenerativeModel("models/gemini-1.5-pro-latest")

@app.route('/ask', methods=['POST'])
def ask_gemini():
    data = request.get_json()
    prompt = data.get('prompt', '')

    if not prompt:
        return jsonify({'error': 'No prompt provided'}), 400

    try:
        response = model.generate_content(prompt)
        return jsonify({'response': response.text})
    except Exception as e:
        return jsonify({'error': str(e)}), 500

if __name__ == '__main__':
    app.run(host='0.0.0.0', port=5000)  # Listens on all interfaces
